Bottom: 6cc8b4d0eeaf70df847e58f3710e9cb5b75e804b
Top:    8cc8c3b36a778681aaf5fe8295deb292cce4df06
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-12-22 21:50:00 +0000

hid/gtk: (GL) Disable double buffering of the GTK drawing widgets

 We already double buffer in GL, so we don't need GTK to "help" us here.
This might speed rendering up a little?


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 23a8490..0467d0e 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -807,6 +807,8 @@ ghid_init_drawing_widget (GtkWidget *widget, GHidPort *port)
 {
   render_priv *priv = port->render_priv;
 
+  GTK_WIDGET_UNSET_FLAGS (GTK_WIDGET (widget), GTK_DOUBLE_BUFFERED);
+
   gtk_widget_set_gl_capability (widget,
                                 priv->glconfig,
                                 NULL,
